### PR TITLE
Triangulation2

### PR DESCRIPTION
**Pull request by dmarsico1**
Fixed the previous pull request from "triangulation" so that it can be merged automatically.

### CODE REVIEW

File: src/FiniteVolumeTools.cpp [hunk 1]
```diff
@@ -628,17 +628,23 @@ void BarycentricCoordinates(
 	double dX3 = nodeV3.x;
 	double dY3 = nodeV3.y;
 	
-	dA = ((dY2 - dY3)*(dX - dX3) + (dX3 - dX2)*(dY - dY3))/
-		 ((dY2 - dY3)*(dX1 - dX3) + (dX3 - dX2)*(dY1 - dY3));
+	double dDenom = (dY2 - dY3)*(dX1 - dX3) + (dX3 - dX2)*(dY1 - dY3);
+	
+	if( abs(dDenom) > 1e-12 ){
```
Comment (paullric) [inline src/FiniteVolumeTools.cpp:633]: Change to fabs(dDenom) > ReferenceTolerance

File: src/FiniteVolumeTools.cpp [hunk 2]
```diff
@@ -244,7 +244,7 @@ void GetTriangleThatContainsPoint(
 	}
 	
 	// First check if point is in initial face
-	if (fTriangleContainsPoint(mesh,iFaceInitial,dX,dY)){
+	if (DoesTriangleContainPoint(mesh,iFaceInitial,dX,dY)){
```
Comment (paullric) [inline src/FiniteVolumeTools.cpp:247]: Is it guaranteed that mesh contains only triangles?  Why use DoesTriangleContainPoint and not DoesFaceContainPoint?

Comment (dmarsico1) [reply src/FiniteVolumeTools.cpp:247]: Yes, this function is only called for the Delaunay triangulation remapping, where mesh is guaranteed to be triangular.

**Comment by rljacob** [comment]
Note from #96 :   "tested using a few idealized test cases (low and high frequency harmonics and a vortex)."

@vijaysm noticed some issues on meshes with holes.

**Pull request merged**